matlab/lib/+Ice/ClassSliceLoader.m

```diff
@@ -0,0 +1,43 @@
+classdef (Sealed) ClassSliceLoader < Ice.SliceLoader
+    % ClassSliceLoader   Implements SliceLoader using an array of meta classes.
+    %
+    % ClassSliceLoader methods:
+    %   ClassSliceLoader - Constructs a ClassSliceLoader from an array of meta classes.
+    %   newInstance - Creates a class or exception instance from a Slice type ID.
+    methods
+        function obj = ClassSliceLoader(metaclassArray)
+            % ClassSliceLoader - Constructs a ClassSliceLoader from an array of meta classes.
+            %
+            % Parameters:
+            %   metaclassArray (matlab.metadata.Class) - An array of matlab.metadata.Class objects.
+
+            obj.typeIdToConstructorMap = configureDictionary('string', 'function_handle');
+
+            for i = 1:length(metaclassArray)
+                mc = metaclassArray(i);
+                if ~isa(mc, 'matlab.metadata.Class')
+                    error('ClassSliceLoader:InvalidArgument', 'Expected array of matlab.metadata.Class.');
+                end
+                for j = 1:length(mc.PropertyList)
```

Comment:
We should throw an exception if the metaclass has no type ID, like in Java:
https://github.com/zeroc-ice/ice/blob/1beb11bd4d15efd5811e4113b4c68d1172decf8f/java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/ClassSliceLoader.java#L35

And we should also check base classes / test base classes.

Reply:
Fixed